### ISSUE TITLE
Create `<UserRoleSelect />`

### ISSUE THREAD

**Issue by aaemnnosttv** [google/site-kit-wp#4821]
## Feature Description

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* 

## Implementation Brief

* 

### Test Coverage

* 

## QA Brief

* 

## Changelog entry

*

**Comment by asvinb** [comment]
@aaemnnosttv Quick question about `UseRoleSelect`. Is only the highlighted selection in the screenshot below:
<img width="721" alt="image" src="https://user-images.githubusercontent.com/31818/164223520-596dcab0-1d7c-483b-a566-40f32ff251b0.png">
because in the AC, it's mentioned:
> The component should be built out according to the [Figma design](https://www.figma.com/file/i04x83uA9U8HHBANlhS7Db/Dashboard-Sharing?node-id=1313%3A25246) with attention paid to both mobile and larger viewports
Right now, in it's current state, and as per the IB, there's nothing related to mobile and larger viewports. It's only when we'll integrate the component within the modal then we'll have to tweak the positioning on small and larger viewports. I can't seem to find another ticket where we actually do the integration.

**Comment by aaemnnosttv** [comment]
@asvinb that makes sense, there might not be much to do responsively for this issue then. The component will be integrated in #4822. Are the chips not slightly larger in mobile?

**Comment by asvinb** [comment]
@aaemnnosttv Correct, the chips are larger on small screens. We should definitely make sure we match the designs but my concern was mostly with the positioning of the close button. I added a note in the IB that we should not take that into account for now and will be taken care of in #4822

**Comment by aaemnnosttv** [comment]
Thanks @asvinb 

> It accepts the following props:
> * `slug`: the module slug

Let's use `moduleSlug` instead to make it clear which slug it is (and not maybe a "role" slug or something like that).

The rest of the IB looks good, although I'm noticing that the ACs are also missing a requirement to update the shared roles when a role is toggled. I'll add this point after which the IB will need updating as well but it should be a small change.

**Comment by aaemnnosttv** [comment]
Thanks @asvinb – the main part of the IB looks good to me. Regarding the tests, I think this component would be a good to add some component tests for, particularly around the behavior of the special "All" choice and the toggling between view/edit modes. We'll probably want to bump the estimate to account for this as well.

**Comment by aaemnnosttv** [comment]
IB ✅

**Comment by wpdarren** [comment]
QA Update: ⚠️ 

@aaemnnosttv this QAB is asking us to check the Storybook but at this point in the engineering we should be checking the functioning on our test site. Do we know what ticket is blocking being able to test this properly?  This is a big piece of functionality and has a lot of testing to go through from the AC.

**Comment by aaemnnosttv** [comment]
@wpdarren this issue is defined as just building the component so there is nowhere else to QA it other than Storybook yet. This will be further QAable when it is integrated with the sharing settings component in https://github.com/google/site-kit-wp/issues/4822.

**Comment by wpdarren** [comment]
### QA Update: ❌ 

@asvinb I have a number of observations listed below:

1. I select a role 'Editor' for Search console and Analytics. Click apply, and the Dashboard Sharing settings modal disappears. I click back on the icon and the two modules I added roles to, are blank, 'no rules'..  I also tested this with AdSense connected and the same issue occurs.  Only seems to be saving PSI and Idea Hub. 

Screencast here:

https://user-images.githubusercontent.com/73545194/171643667-63050c39-82f5-489f-9f4b-05430014f6fa.mp4

2. The 'Learn more' link doesn't have the external link icon. 

3.  I only had one admin user when I initially added roles to two modules. I then created a secondary admin user and went back to the `Dashboard sharing & permissions` screen, and the `Who can manage view access` dropdown is inactive for those modules. They are active for the two modules that weren't previously set. Maybe this is expected but even though these are shareable modules, we should allow the just the connected admin permission?  

![image](https://user-images.githubusercontent.com/73545194/171641913-04f6b82f-3933-4745-a6b7-ac9051d5c509.png)

4. Odd UX/UI in a specific scenario.  I go to a module, i.e. Idea Hub, and add any role, and click on the apply button. It's the only change that I want to make so do not click on the X icon to close it.  When I go back to `Dashboard sharing & permissions` screen, the Idea Hub, shows the roles, because I didn't close it before applying.  It would be more user friendly, to have the roles close when the apply button is clicked.  Here's a screencast. 

https://user-images.githubusercontent.com/73545194/171646262-a716a768-60c3-4684-8074-2053eb423243.mp4

**Note:** the same behaviour appears when you also click on the 'Cancel' link. 

5. For the `Product` and `Who can view` titles, the font is Google Sans on Figma but is Roberto on my site. The font sizes and color are slightly different too between the site and Figma. This could be expected but did not want to assume. 

![image](https://user-images.githubusercontent.com/73545194/171653792-df744287-79dc-4dd4-9ed8-1333dc1fb961.png)

6. For the module titles and roles, the font is Google Sans on Figma but is Roberto on my site. The font colors are also different when viewing the different texts (module title, role title and 'Add role' text, as per screenshot. 

![image](https://user-images.githubusercontent.com/73545194/171655528-5ecdd2ab-ed11-480b-b64e-63acf92df381.png)

7. The closing of the Dashboard Sharing settings modal when clicking outside is inconsistent. When clicking in some areas I am able to close it, but other areas I am not.  Seems that you can close it when clicking above or below the modal but not at the side, not a huge issue, but thought I would mention it as it's likely to come up in bug bash. 

https://user-images.githubusercontent.com/73545194/171654543-7a400d84-d51d-4a88-aeff-798269b9e90f.mp4

8. Minor UX/UI on the Desktop Sharing settings icon in the header.  When the modal is open, the icon does not have any hover/active background color, which is to be expected.  When you click cancel or apply in the modal, it closes as expected but the icon is now highlighted. The colour that the icon is highlighted is a different colour than when you hover over it.

![image](https://user-images.githubusercontent.com/73545194/171657213-2b8dfc20-1bc3-48a9-a026-cffcd9a06eb1.png)

**Comment by asvinb** [comment]
@wpdarren As per our conversation, let's consolidate all our QA findings in #4822 and close this one. Thanks!
